silx/io/dictdump.py

```diff
@@ -207,35 +207,53 @@ def dicttoh5(treedict, h5file, h5path='/',
         h5path += "/"
 
     with _SafeH5FileWrite(h5file, mode=mode) as h5f:
+
+        # The root of the tree can be a list or tuple of dictionaries.
+        if (isinstance(treedict, (list, tuple))
+                and all(isinstance(item, dict) for item in treedict)):
+            for i, item in enumerate(treedict):
+                dicttoh5(item, h5f, h5path + '/%d' % i,
```

Comment:
To make it sorted, maybe you want to use 0 prefixed indices with something like: `ndigits = int(numpy.log10(len(treedict))) + 1`